test/unittest/blas2/blas2_trsv_test.cpp

```diff
@@ -0,0 +1,139 @@
+/***************************************************************************
+ *
+ *  @license
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename blas2_trsv_test.cpp
+ *
+ **************************************************************************/
+
+#include "blas_test.hpp"
+
+template <typename T>
+using combination_t =
+    std::tuple<index_t, bool, bool, bool, index_t, index_t, T>;
+
+template <typename scalar_t>
+void run_test(const combination_t<scalar_t> combi) {
+  index_t n;
+  bool trans;
+  bool is_upper;
+  bool is_unit;
+  index_t incX;
+  index_t lda_mul;
+  scalar_t wa;
+  std::tie(n, is_upper, trans, is_unit, incX, lda_mul, wa) = combi;
+
+  const char* t_str = trans ? "t" : "n";
+  const char* uplo_str = is_upper ? "u" : "l";
+  const char* diag_str = is_unit ? "u" : "n";
+
+  index_t a_size = n * n * lda_mul;
+  index_t x_size = 1 + (n - 1) * incX;
+
+  // Input matrix
+  std::vector<scalar_t> a_m(a_size);
+  // Input/output vector
+  std::vector<scalar_t> x_v(x_size);
+  // Input/output system vector
+  std::vector<scalar_t> x_v_cpu(x_size);
+
+  // Control the magnitude of extra-diagonal elements
+  for (index_t i = 0; i < n; ++i)
+    for (index_t j = 0; j < n; ++j)
+      a_m[(j * n * lda_mul) + i] =
+          ((!is_upper && (i > j)) || (is_upper && (i < j)))
+              ? random_scalar(scalar_t(-10), scalar_t(10)) / scalar_t(n)
+              : NAN;
+
+  if (!is_unit) {
+    // Populate main diagonal with dominant elements
+    for (index_t i = 0; i < n; ++i)
+      a_m[(i * n * lda_mul) + i] = random_scalar(scalar_t(9), scalar_t(11));
```

Comment:
Do we need to specify the range to be between `9` and `11`?

Reply:
Basically, we need to control the magnitude of the random numbers in order to produce a linear system that has a solution.